sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/PropertyGetterAndSetterUsageCheck.kt

```diff
@@ -0,0 +1,157 @@
+                val getterIdentifier = findJavaStyleGetterFuncIdentifier(propIdentifier.text, propType, javaAccessors, bindingCtx)
+                val setterIdentifier = findJavaStyleSetterFuncIdentifier(propIdentifier.text, propType, javaAccessors, bindingCtx)
+                if (getterIdentifier != null) {
+                    val secondaries = when (setterIdentifier) {
+                        null -> listOf(ctx.secondaryOf(propIdentifier, """Property "${propIdentifier.text}""""))
+                        else -> listOf(
+                            ctx.secondaryOf(propIdentifier, """Property "${propIdentifier.text}""""),
+                            ctx.secondaryOf(setterIdentifier, """Setter to convert to a "set(value)"""")
+                        )
+                    }
+                    ctx.reportIssue(
+                        getterIdentifier,
+                        """Convert this getter to a "get()" on the property "${propIdentifier.text}".""",
+                        secondaries
+                    )
+                } else if (setterIdentifier != null) {
+                    val secondaries = listOf(ctx.secondaryOf(propIdentifier, """Property "${propIdentifier.text}""""))
+                    ctx.reportIssue(
+                        setterIdentifier,
+                        """Convert this setter to a "set(value)" on the property "${propIdentifier.text}".""",
+                        secondaries
+                    )
+                }
+            }
+        }
+    }
+
+    companion object {
+        private fun isPrivateWithoutGetterAndSetter(prop: KtProperty): Boolean = prop.isPrivate() && prop.accessors.isEmpty()
+
+        private fun isGetterOrSetter(parameterCount: Int, functionName: String) =
+            (parameterCount == 0 && GETTER_PREFIX.find(functionName) != null) ||
+                (parameterCount == 1 && SETTER_PREFIX.find(functionName) != null)
+
+        private fun findJavaStyleGetterFuncIdentifier(
+            propName: String, propType: KotlinType, javaAccessors: Map<String, List<KtNamedFunction>>, bindingCtx: BindingContext
+        ): PsiElement? {
+            val capitalizedName = capitalize(propName)
+            val functionsPrefixedByIs = if (propType.matches("kotlin.Boolean")) {
+                javaAccessors.getOrElse("is${capitalizedName}") { emptyList() }
+            } else {
+                emptyList()
+            }
+            val matchingFunctions = (javaAccessors.getOrElse("get${capitalizedName}") { emptyList() } + functionsPrefixedByIs)
+                .filter { it.returnType(bindingCtx) == propType }
+            return unambiguousIdentifier(matchingFunctions)
+        }
+
+        private fun parameterMatchesType(parameter: KtParameter, type: KotlinType, bindingCtx: BindingContext): Boolean {
+            return !parameter.isVarArg && type == (parameter.typeReference?.let { bindingCtx.get(BindingContext.TYPE, it) })
```

Comment:
Here the `==` operator is used to compare `KotlinType`s. One from `KtProperty#typeReference` and one from `KtParameter#typeReference`. I don't know if it's possible to do this but the underlying logic seems to be what I want:
```kotlin
isMarkedNullable == other.isMarkedNullable && StrictEqualityTypeChecker.strictEqualTypes(unwrap(), other.unwrap())
```

Reply:
I want to compare the type of X with Y and Z in this code:
```kotlin
var value: X
fun getValue(): Y
fun setValue(value: Z)
```
For example when X is `List<String>?`, I want to say that Y and Z match X when:
- They have the same type which is `kotlin.collections.List`, this can be done by comparing `getJetTypeFqName(false)`, it's what `FunMatcher` uses.
- They have the same type arguments, which is `<kotlin.String>`, `KotlinType#arguments` or `getJetTypeFqName(true)`. (this is not supported by `FunMatcher`)
- They have the same nullability. `KotlinType#isMarkedNullable` (this is not supported by `FunMatcher`)

I notice that `==` compares all what I want, but it could also be too accurate for my need. For example, perhaps you could state that we can ignore the `TypeArgumentMarker.getVariance()` comparison. That's why I'm asking.